inputremapper/migrations.py

```diff
@@ -142,6 +142,37 @@ def _rename_config():
         shutil.move(old_config_path, CONFIG_PATH)
 
 
+def _add_target():
+    """add the target field to each preset mapping
+
+    Always use "keyboard" as target. This will break mappings which map codes not in the capabilities of keyboard.
+    We can not check code names or capabilities of the available targets due to circular imports
```

Comment:
Any Idea if we can do something about this? We currently can not access `system_mapping` so there is no good way to find a target with the right capabilities for the mapping.
also I am not sure if we can import `global_uinputs` and `parse`

Reply:
why can't we import global_uinputs? Where is the circular import?

Reply:
seriously, if you just check if there is a BTN_ written somewhere in the mapping and use gamepad instead for that it's going to be fine probably

Reply:
No, i think that would work. The problem is that we can not import system_mapping, DEFAULT_UINPUTS and global_uinputs only contain the key-codes, not all the different key-names which are in the preset files?

Reply:
ah right.

how about something like

```
>>> import evdev
>>> evdev.ecodes.ecodes["KEY_A"]
30

Reply:
If `migrate()` is called in `bin/input-remapper-gtk` and `bin/input-remapper-control` instead of `config` I think there won't be a circular import anymore

Reply:
how does the autoloading process work exactly? Can we be sure migrate() is called before any presets autoload?

Reply:
autoloading uses input-remapper-control: https://github.com/sezanzeb/input-remapper/blob/main/data/99-input-remapper.rules

Just make sure to import it early and call it before importing the other modules.

Reply:
seems to work now